models/Article.php

```diff
@@ -106,108 +95,217 @@ function get_post(){
-        $req->execute($p);
-
-        $id = $db->lastInsertId(); //On doit mettre la fonction lastInsertId seulement après une fonction INSERT.
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
     }
 
     // Fonction permettant d'insérer une nouvelle image dans la base de données
     function post_img($tmp_name, $extension){
-        global $db;
-        global $id;
-        $db = getPdo();
+        $id = $this->db->lastInsertId(); //On doit mettre la fonction lastInsertId seulement après une fonction INSERT.
         
-      
-        $i = [
+        $tableau = [
             'id'    =>  $id,
             'image' =>  $id.$extension  
         ];
 
         $sql = "UPDATE articles SET image = :image WHERE id = :id";
         
-        $req = $db->prepare($sql);
-        $req->execute($i);
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
         move_uploaded_file($tmp_name, "public/img/posts/".$id.$extension);
     }
 
         // Fonction permettant de mettre à jour l'image d'un article
     function update_img($tmp_name, $extension){
-        $db = getPdo();
-        
-        $id = $_GET['id'];
+        $id = filter_input(INPUT_GET, 'id', FILTER_SANITIZE_NUMBER_INT);
          
-        $i = [
+        $tableau = [
             'id'    =>  $id,
             'image' =>  $id.$extension  
         ];
 
         $sql = "UPDATE articles SET image = :image WHERE id = :id";
         
-        $req = $db->prepare($sql);
-        $req->execute($i);
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
         move_uploaded_file($tmp_name, "public/img/posts/".$id.$extension);
     }
     
     // Fonction permettant de supprimer un article
     function delete_article()
-    {
-        $db = getPdo();
+    {  
+        $id = filter_input(INPUT_GET, 'id', FILTER_SANITIZE_NUMBER_INT);
         
-        if (empty($_GET['id']) || !ctype_digit($_GET['id'])) {
-            die("Ho ! Fallait préciser le paramètre id en GET !");
-        }
-
-        $id = $_GET['id'];
+        $query = $this->db->prepare('
+        DELETE
+        FROM articles 
+        WHERE id = :id');
+        $query->execute(['id' => $id]
+        );
+    }
+    
+      // Fonction permettant de supprimer les commentaires d'un article
+    function delete_article_comments()
+    {  
+        $id = filter_input(INPUT_GET, 'id', FILTER_SANITIZE_NUMBER_INT);
 
-        $query = $db->prepare('SELECT * FROM articles WHERE id = :id');
+        $query = $this->db->prepare('
+        DELETE 
+        FROM comments 
+        WHERE article_id = :id
+        ');
         $query->execute(['id' => $id]);
-        if ($query->rowCount() === 0) {
-            die("Aucun commentaire n'a l'identifiant $id !");
+
+        header("Location: /liste-de-tous-les-articles");
+    }
+    
+    //Fonction permettant de vérifier les données envoyées par l'utilsiateur
+     function form_page_write(){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [form_page_write accesses the super-global variable $_FILES.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5344015)